woodwork/table_accessor.py

```diff
@@ -129,6 +129,21 @@ def schema(self):
         if self._schema:
             return self._schema._get_subset_schema(list(self.columns.keys()))
 
+    def set_index(self, new_index):
+        """Sets the index column of the DataFrame. Adds the 'index' semantic tag to the column
+        and clears the tag from any previously set index column.
+        Setting a column as the index column will also cause any previously set standard
+        tags for the column to be removed.
+
+        Args:
+            index (str): The name of the column to set as the index
+        """
+        self._schema.set_index(new_index)
+
+        if self._schema.index is not None:
+            _check_index(self._dataframe, self._schema.index)
+        self._set_underlying_index()
```

Comment:
@thehomebrewnerd here's an example of Schema/Accessor separation: Setting the index on the schema and then using that updated schema to change the dataframe. 

We don't have to keep it like this, but I thought I'd show an example of it.

Reply:
So this pattern makes me think about a scenario where the _check_index validation might fail (say a column is not unique). We will have already set the index in the schema to an invalid column and it will remain that way. Not suggesting we need to fix at this point, because presumably, the user would then do something to fix the error from _check_index, which could then resync the information.

Just something to think about I guess.